api/routes/discovery.py

```diff
@@ -1,141 +1,174 @@
+            must.append({'text': {'path': 'isPartOf.@list.name', 'query': self.isPartOfName}})
+        if self.associatedMediaName:
+            must.append({'text': {'path': 'associatedMedia.@list.name', 'query': self.associatedMediaName}})
+        if self.fundingGrantName:
+            must.append({'text': {'path': 'funding.@list.grant.name', 'query': self.fundingGrantName}})
+        if self.fundingFunderName:
+            must.append({'text': {'path': 'funding.@list.funder.name', 'query': self.fundingFunderName}})
+
+        return must
+
+    @property
+    def stages(self):
+        highlightPaths = ['name', 'description', 'keywords', 'creator.@list.name']
+        stages = []
+        if self.dataCoverageStart or self.dataCoverageEnd:
+            stages.append(
+                {
+                    '$addFields': {
+                        'dataCoverageDates': {
+                            '$map': {
+                                'input': {'$split': ['$dataCoverage', '/']},
+                                'in': {'$dateFromString': {'dateString': '$$this', 'onError': datetime.utcnow()}},
+                            }
+                        }
+                    }
+                }
+            )
+
+        stages.append(
+            {
+                '$search': {
+                    'index': 'fuzzy_search',
+                    'compound': {'filter': self._filters, 'should': self._should, 'must': self._must},
+                    'highlight': {'path': highlightPaths},
                 }
             }
-        })
+        )
 
-    # Sort needs to happen before pagination
-    if sortBy:
-        stages.append({
-            '$sort': {
-                sortBy: 1
-            }
-        })
+        if self.clusters:
+            stages.append({'$match': {'clusters': {'$all': self.clusters}}})
 
-    stages.append(
-        {
-            '$skip': (pageNumber - 1) * pageSize
-        }
-    )
-    stages.append(
-        {
-            '$limit': pageSize
-        },
-    )
-    stages.append(
-        {
-            '$unset': ['_id']
-        }
-    )
-    stages.append(
-        {
-            '$set': {
-                'score': {'$meta': 'searchScore'},
-                'highlights': {'$meta': 'searchHighlights'}
-            }
-        },
-    )
+        # sorting needs to happen before pagination
+        if self.sortBy:
+            stages.append({'$sort': {self.sortBy: 1}})
+        stages.append({'$skip': (self.pageNumber - 1) * self.pageSize})
+        stages.append({'$limit': self.pageSize})
+        stages.append({'unset': ['_id']})
+        stages.append(
+            {'$set': {'score': {'$meta': 'searchScore'}, 'highlights': {'$meta': 'searchHighlights'}}},
+        )
+        return stages
 
-    result = await request.app.mongodb["cznet"].aggregate(stages).to_list(pageSize)
+
+@router.get("/search")
+async def search(request: Request, search_query: SearchQuery):
+    stages = search_query.stages
```

Comment:
Somebody spent some time in the documentation for pydantic.  Thanks for these changes.